pod/recorder/views.py

```diff
@@ -371,12 +372,17 @@ def studio_pod(request):
         )
     # Render the Opencast studio index file
     opencast_studio_rendered = render_to_string("studio/index.html")
-    # head = opencast_studio_rendered[opencast_studio_rendered.index("<head>")
-    # + len("<head>"):opencast_studio_rendered.index("</head>")]
+    head = opencast_studio_rendered[
+        opencast_studio_rendered.index("<head>")
+        + len("<head>"):opencast_studio_rendered.index("</head>")
+    ]
+    scripts = re.findall('<script .[a-z="]+ src=".[a-z/.0-9]+"></script>', head)
+    styles = re.findall('<style>.*</style>', head)
```

Comment:
Cette partie risque de déclancher une exception sur une version incompatible d'OpenCast.
Je propose de la mettre en try ainsi :

```python
try:
        head = opencast_studio_rendered[
            opencast_studio_rendered.index("<head>")
            + len("<head>"):opencast_studio_rendered.index("</head>")
        ]
        scripts = re.findall('<script .[a-z="]+ src=".[a-z/.0-9]+"></script>', head)
        styles = re.findall('<style>.*</style>', head)
    except ValueError as e:
        if "substring not found" in str(e):
            messages.add_message(
                request,
                messages.ERROR,
                _("Error in Esup-Pod Opencast rendering. Opencast must probably be updated."),
            )
            scripts = styles = []
        else:
            raise e
```

Reply:
hello, sauf erreur de ma part findall ne leve pas d'exception s'il ne trouve pas de donnée. par contre, s'il ne trouve pas head, la oui mais bon, dans une page html, il y a tjs un head

Reply:
par contre, on pourrait utiliser une lib comme beautifulsoup pour faire la mm chose

Reply:
Sur mon test, j'ai eu cette erreur ("substring not found" : head)

Reply:
oui, j'ai fixé ce probleme en rajoutant head dans le html du test unitaire

Reply:
Mon fichier index.html contenait uniquement ```html
        <html>
            <body>
                <h1>Heading</h1>
            </body>
        </html>
```
Mais j'avoue que je ne sais pas pourquoi. Je viens de recompiler studio et il est bien rempli cette fois (je suis sur la version de nov 2022 pour vérifier la compatibilité)

Reply:
J'ai trouvé d'où venait ce index.html : c'est Pod qui le crée quand il fait des tests. Il faudrait ptet ne l'écraser que s'il n'existe pas déja ? en tout cas cela ne concerne que les développeur, donc on peut ne pas en tenir compte.